ptf/saibuffer.py

```diff
@@ -0,0 +1,671 @@
+
+        self.assertGreater(bp_curr_occupancy_bytes, 0)
+        self.assertGreaterEqual(
+            stats["SAI_BUFFER_POOL_STAT_WATERMARK_BYTES"], expected_watermark)
+
+        stats = sai_thrift_get_ingress_priority_group_stats(
+            self.client, self.ipg)
+
+        print("SAI_INGRESS_PRIORITY_GROUP_STAT_CURR_OCCUPANCY_BYTES "
+              "(max measured)", ipg_curr_occupancy_bytes)
+        print("SAI_INGRESS_PRIORITY_GROUP_STAT_SHARED_CURR_OCCUPANCY_BYTES "
+              "(max measured)", ipg_shared_curr_occupancy_bytes)
+        print("SAI_INGRESS_PRIORITY_GROUP_STAT_WATERMARK_BYTES",
+              stats["SAI_INGRESS_PRIORITY_GROUP_STAT_WATERMARK_BYTES"])
+        print("SAI_INGRESS_PRIORITY_GROUP_STAT_PACKETS",
+              stats["SAI_INGRESS_PRIORITY_GROUP_STAT_PACKETS"])
+        print("SAI_INGRESS_PRIORITY_GROUP_STAT_BYTES",
+              stats["SAI_INGRESS_PRIORITY_GROUP_STAT_BYTES"])
+        print("SAI_INGRESS_PRIORITY_GROUP_STAT_DROPPED_PACKETS",
+              stats["SAI_INGRESS_PRIORITY_GROUP_STAT_DROPPED_PACKETS"])
+
+        self.assertGreater(ipg_curr_occupancy_bytes, 0)
+        self.assertGreater(ipg_shared_curr_occupancy_bytes, 0)
+        self.assertGreaterEqual(
+            stats["SAI_INGRESS_PRIORITY_GROUP_STAT_WATERMARK_BYTES"],
+            expected_watermark)
+        self.assertEqual(
+            stats["SAI_INGRESS_PRIORITY_GROUP_STAT_PACKETS"], self.tx_cnt)
+        self.assertEqual(
+            stats["SAI_INGRESS_PRIORITY_GROUP_STAT_BYTES"],
+            self.tx_cnt * self.pkt_len)
+        self.assertEqual(
+            stats["SAI_INGRESS_PRIORITY_GROUP_STAT_DROPPED_PACKETS"],
+            expected_drops)
+
+    def clearVerify(self):
+        """
+        Clear bufer pool and ingress priority group stats.
+        Verify they are cleared.
+        """
+        print()
+        print("Clear bufer pool and ingress priority group stats")
+
+        status = sai_thrift_clear_buffer_pool_stats(
+            self.client, self.ingr_pool)
+        self.assertEqual(status, SAI_STATUS_SUCCESS)
+
+        status = sai_thrift_clear_ingress_priority_group_stats(
+            self.client, self.ipg)
+        self.assertEqual(status, SAI_STATUS_SUCCESS)
+
+        print("Get stats and verify they are cleared")
+
+        stats = sai_thrift_get_buffer_pool_stats(self.client, self.ingr_pool)
+
+        self.assertEqual(
+            stats["SAI_BUFFER_POOL_STAT_WATERMARK_BYTES"], 0)
+
+        stats = sai_thrift_get_ingress_priority_group_stats(
+            self.client, self.ipg)
+
+        self.assertEqual(
+            stats["SAI_INGRESS_PRIORITY_GROUP_STAT_PACKETS"], 0)
+        self.assertEqual(
+            stats["SAI_INGRESS_PRIORITY_GROUP_STAT_BYTES"], 0)
+        self.assertEqual(
+            stats["SAI_INGRESS_PRIORITY_GROUP_STAT_WATERMARK_BYTES"], 0)
+        self.assertEqual(
+            stats["SAI_INGRESS_PRIORITY_GROUP_STAT_DROPPED_PACKETS"], 0)
+
+    def runTest(self):
+        print()
```

Comment:
Are these empty prints at the beginning of functions needed? Same for the other functions.

Reply:
For readability reason during test execution - maybe more Pythonic than "\n" - a matter of taste.